### PR TITLE
CLC-5052 Expanding the functionality of Mockable

### PR DESCRIPTION
**Pull request by ctweney**
https://jira.ets.berkeley.edu/jira/browse/CLC-5052

3 new proxies are now on the Mockable framework. Mockable now correctly handles basic authentication. It also provides a read_file convenience method. And, you can now mock proxies that deliver XML instead of JSON. 

Note: This change requires a settings.yml configuration change: Renaming the feed_url param to base_url in 3 proxies: blog_latest_release_notes_feed_proxy, app_alerts_proxy, cal1card_proxy. Changing the setting will be part of the release notes.

### CODE REVIEW

**Comment by paulkerschen** [comment]
@ctweney , this looks awesome, thank you. Just waiting on Travis.

File: lib/proxies/mockable.rb [hunk 1]
```diff
@@ -38,5 +39,9 @@ def mock_json
       ''
     end
 
+    def read_file(*args)
+      File.read(Rails.root.join(*args))
+    end
+
```
Comment (paulkerschen) [inline lib/proxies/mockable.rb]: +1

**Comment by paulkerschen** [comment]
@ctweney Also, should we update the YAML on calcentral-dev concurrently with merging this? That's one task I haven't yet had to manage.

**Comment by ctweney** [comment]
@paulkerschen  Yes, ideally the YML update on dev should happen immediately after merge. Please ping me on IM and I can take care of it for you.

**Pull request merged**